src/components/Main.js

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from 'react';
+import { connect, useSelector, useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { setMonsters, getFiltered, receiveMonsterDetails } from '../redux/monsters';
+import MonsterCard from './MonsterCard';
+import getMonsters from '../api';
+
+const MainMonstersList = (props) => {
+  const dispatch = useDispatch();
+  const [filter, setFilter] = useState('');
+  const monstersList = useSelector((state) => state.monsterReducer.monsters);
+  const monsters = async () => {
+    const potato = await getMonsters();
+    dispatch(setMonsters(potato));
+  };
+  useEffect(() => {
```

Comment:
- [ ] OPTIONAL - Consider adding a loading indicator so it doesn't look like the page is broken when retrieving data.